### DISCUSSION TITLE
lspsaga not applying colors to the icons in winbar

### DISCUSSION THREAD

**Discussion by HendrikPetertje** [nvimdev/lspsaga.nvim#606] [Q&A] [answered]
Hey all, I recently updated lspsaga and I am having a blast with all the new stuff!
there is one little problem i am dealing with however. 

I'm trying to get colors working in the winbar and I see that they are there when i do an inspect on 
`lspsaga.get_symbol_node()` as you can see below:

<img width="899" alt="image" src="https://user-images.githubusercontent.com/3583395/203634172-2867cb93-f38a-4fec-9fb2-cee203a1f157.png">

and I have setup colors like this in my config:

```
local function get_palette()
  if vim.g.colors_name == "catppuccin" then
    -- If the colorscheme is catppuccin then use the palette.
    return require("catppuccin.palettes").get_palette()
  else
    -- Default behavior: return lspsaga's default palette.
    local palette = require("lspsaga.lspkind").colors
    palette.peach = palette.orange
    palette.flamingo = palette.orange
    palette.rosewater = palette.yellow
    palette.mauve = palette.violet
    palette.sapphire = palette.blue
    palette.maroon = palette.orange

    return palette
  end
end

local colors = get_palette()

lspsaga.init_lsp_saga({
  ...
  custom_kind = {
    File = { " ", colors.rosewater },
    Module = { " ", colors.blue },
    Namespace = { " ", colors.blue },
    Package = { " ", colors.blue },
    Class = { "ﴯ ", colors.yellow },
    Method = { " ", colors.blue },
    Property = { "ﰠ ", colors.teal },
    Field = { " ", colors.teal },
    Constructor = { " ", colors.sapphire },
    Enum = { " ", colors.yellow },
    Interface = { " ", colors.yellow },
    Function = { " ", colors.blue },
    Variable = { " ", colors.peach },
    Constant = { " ", colors.peach },
    String = { " ", colors.green },
    Number = { " ", colors.peach },
    Boolean = { " ", colors.peach },
    Array = { " ", colors.peach },
    Object = { " ", colors.yellow },
    Key = { " ", colors.red },
    Null = { "ﳠ ", colors.yellow },
    EnumMember = { " ", colors.teal },
    Struct = { " ", colors.yellow },
    Event = { " ", colors.yellow },
    Operator = { " ", colors.sky },
    TypeParameter = { " ", colors.maroon },
    -- ccls-specific icons.
    TypeAlias = { " ", colors.green },
    Parameter = { " ", colors.blue },
    StaticMethod = { "ﴂ ", colors.peach },
    Macro = { " ", colors.red },
  },
  ...
})
```

But my bar looks like this so it feels like something is going wrong

<img width="304" alt="image" src="https://user-images.githubusercontent.com/3583395/203635418-f4c5c9bd-b0f7-4aae-90db-2d3ed1d3b368.png">

Am I doing something wrong, or is there something amiss in the plugin itself?

**Comment by HendrikPetertje** [comment]
This is my full config:
https://git.hendrikpeter.net/hendrikpeter/pico-vim/-/blob/master/lua/peva/lspsaga.lua

edit: I have gone ahead and written a little regex to get rid of the tags for now, as the lsp saga stuff was changing the colour of my lualine winbar.

**Comment by glepnir** [comment, marked as answer]
test with a min config it works fine.

![image](https://user-images.githubusercontent.com/41671631/203755282-38c8080a-0087-4dda-8917-b8830ff20705.png)

**Comment by HendrikPetertje** [comment]
Cleaned up a bit, removed the custom color definitions and now it seems to be working!
